oauthproxy.go

```diff
@@ -1010,7 +1012,7 @@ func prepareNoCacheMiddleware(next http.Handler) http.Handler {
 // This is usually the OAuthProxy callback URL.
 func (p *OAuthProxy) getOAuthRedirectURI(req *http.Request) string {
 	// if `p.redirectURL` already has a host, return it
-	if p.redirectURL.Host != "" {
+	if p.relativeRedirectURL || p.redirectURL.Host != "" {
```

Comment:
I was wondering if we can somehow test this in a unit test? 🤔 We do have a test in `oauthproxy_test.go` regarding redirect from the SignPage `TestSignInPageIncludesTargetRedirect`

Reply:
I was not really able to understand fully how to change TestSignInPageIncludesTargetRedirect, instead I create a test for the getOAuthRedirectURI functions with different config for the OAuthProxy.